apps/src/JSInterpreter.js

```diff
@@ -84,8 +86,20 @@ JSInterpreter.prototype.parse = function (options) {
     codegen.createNativeFunctionFromInterpreterFunction = function (intFunc) {
       return function () {
         if (self.initialized()) {
-          self.queueEvent(intFunc, arguments);
-          
+
+          // Convert arguments to array in fastest way possible and avoid
+          // "Bad value context for arguments variable" de-optimization
+          // http://jsperf.com/array-with-and-without-length/5
+          var args = new Array(arguments.length);
+          for (var i = 0; i < arguments.length; i++) {
+            args[i] = arguments[i];
+          }
+
+          self.eventQueue.push({
+            'fn': intFunc,
+            'arguments': args
+          });
```

Comment:
note, quotes are unneeded in keys unless they have spaces or dashes, i.e. I would prefer
```
self.eventQueue.push({
  fn: initFunc,
  arguments: args
});